test pull request

J35P312 opened the pull request.

J35P312 committed:
Update AUTHORS

J35P312 committed:
modified:   src/FindTranslocations.cpp

vezzi commented on src/FindTranslocations.cpp:
```diff
@@ -15,9 +16,127 @@
+	string tmpstr;
+	if(arguments == 6){
+		try{
+			for(int i=0;i<=1;i++){
```
This is strange, the for loop seems not needed as you iterate it at most once being the condition i<=1

vezzi commented on src/FindTranslocations.cpp:
```diff
@@ -15,9 +16,127 @@
+			}
+			//if the ID of the chromosome is given as a number, the refID is a number, if the X or Y chromosome is given, the correct refID must be found.
+			if(isNumber==1){
+				chr[i] = atoi(tmpstr.c_str());
```
much better if you use chr2pos and pos2chr that I create, this solution is not general enough:
- there are many others "chromosomes" in the human genome (unplaced contigs, unknowns, mitochondrial) 
- there is no reason to lock the tool to the human genome, if you are general enough this tool can be used in all available genomes

vezzi commented on src/FindTranslocations.cpp:
```diff
@@ -15,9 +16,127 @@
+
+
+	string tmpstr;
+	if(arguments == 6){
```
This is in general is not recommended, if you have many of these checks in a function makes the code difficult to maintain. Imagine that in three months I want to add a new variable, at this point I need to increment 6 to 7

vezzi commented on src/FindTranslocations.cpp:
```diff
@@ -15,9 +16,127 @@
+			}
+			
+			}
+		}catch(...){
```
Good that you use try and catch, I never do it I should learn!!!!

vezzi commented on src/FindTranslocations.cpp:
```diff
@@ -15,9 +16,127 @@
+
+
+	//collects all reads found in the given intervals and writes them to the file outputFileHeader+"_roi.bam"
+	for(int i=0;i<=1;i++){
```
Also in this case the for loop is incomplete, maybe this is a test

vezzi commented on src/FindTranslocations.cpp:
```diff
@@ -79,119 +199,128 @@ int main(int argc, char *argv[]) {
 
-	if (vm.count("max-insert")) {
-		max_insert = vm["max-insert"].as<int>();
+	if(vm.count("extract")){
```
Maybe the code needs some reorganisation... the program might have run modalities. --extract extracts from the bam file regions, --struct-var looks for structural variations, --copy-number looks for copy number variations, and the modalities are exclusive, i.e., you can run the program only in one of the modalities (like it happens for samtools)

vezzi commented on src/FindTranslocations.cpp:
```diff
@@ -54,7 +174,7 @@ int main(int argc, char *argv[]) {
 						("insert",       po::value<float>(), "do not compute insert size from bam file, use the one specified here (must be used in combination with --coverage --insert-std)")
 						("insert-std",   po::value<float>(), "do not compute insert size standard deviation from bam file, use the one specified here (must be used in combination with --insert --coverage)")
-						;
+						("extract",       po::value<string>(), "extracts the reads involved in an event, ""CHRA startPosA endPosA CHRB startPosB endPosB"" ");
```
Maybe the input should be the output file from FindTranslocation or similar to allow one to extract more than one event

vezzi commented on src/FindTranslocations.cpp:
```diff
@@ -15,9 +16,127 @@
 
 #include "data_structures/Translocation.h"
 #include  <boost/program_options.hpp>
+#include "api/BamWriter.h"
```
it is likely that you need to add this file to the commit...

J35P312 committed:
modified:   src/FindTranslocations.cpp

J35P312 committed:
modified:   src/FindTranslocations.cpp
	modified:   src/data_structures/Translocation.cpp
	modified:   src/data_structures/Translocation.h
